### ISSUE TITLE
DesktopVK shouldn't require Vulkan SDK installed for release builds & client devices

### ISSUE THREAD

**Issue by squarebananas** [MonoGame/MonoGame#8535]
### Prerequisites

- [X] I have verified this issue is present in the `develop` branch
- [X] I have searched [open](https://github.com/MonoGame/MonoGame/issues) and [closed](https://github.com/MonoGame/MonoGame/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported.

### MonoGame Version

MonoGame 3.8.2.1-develop

### Which MonoGame platform are you using?

N/A

### Operating System

Windows

### Description

Currently a DesktopVK game will crash if the client device doesn't have the Vulkan SDK installed. A workaround is to comment out this line:
https://github.com/MonoGame/MonoGame/blob/97c029b72c1bbd853d147c89301d36c6f3b3e394/native/monogame/vulkan/MGG_Vulkan.cpp#L651
Just noticed `CheckValidationLayerSupport` returns a unused `uint64_t`. I think that may have been intended to remove unavailable layer names prior to `vkCreateInstance`.

### Steps to Reproduce

Run a game built with DesktopVK on a device without Vulkan SDK.

### Expected Behavior

Should run

### Resulting Behavior

Don't run

**Comment by mrhelmut** [comment]
Yes, the results from ```CheckValidationLayerSupport()``` need to be used to remove unsupported layers from ```enabledLayers``` (or doing it directly inside the function. Otherwise it will indeed try to enable them on unsupported setup.

Secondly, we also need to enable them only in DEBUG builds, otherwise whoever with a vulkan SDK installed will have the validation layers enabled without knowing (and validation layers impair performances) in release builds (e.g. when playing a game casually instead of developing).

**Comment by ThomasFOG** [comment]
Fixed locally.
